timesketch/api/v1/resources/sigma.py

```diff
@@ -87,15 +125,274 @@ def get(self, rule_uuid):
+    """Resource to get list of SigmaRules."""
+
+    @login_required
+    def get(self):
+        """Handles GET request to the resource.
+
+        Returns:
+            Dict of sigma rules
+        """
+        sigma_rules = []
+
+        all_sigma_rules = SigmaRule.query.all()
+        # Return a subset of the Sigma objects to reduce the amount of
+        # data sent to the client.
+        for rule in all_sigma_rules:
+            sigma_rules.append(_enrich_sigma_rule_object(rule=rule))
+
+        meta = {"rules_count": len(sigma_rules)}
+        return jsonify({"objects": sigma_rules, "meta": meta})
+
+
+class SigmaRuleResource(resources.ResourceMixin, Resource):
+    """Resource to get / delete / post / put a Sigma rule."""
+
+    @login_required
+    def get(self, rule_uuid):
+        """Handles GET request to the Sigma Rule resource.
+
+        Handels GET API calls to /sigmarule/<string:rule_uuid>/ where the
+        rule_uuid is the primaray way to identify the rule.
+
+        Args:
+            rule_uuid: uuid of the rule
+
+        Returns:
+            JSON sigma rule
+        """
+        try:
+            rule = SigmaRule.query.filter_by(rule_uuid=rule_uuid).first()
+
+        except Exception as e:  # pylint: disable=broad-except
+            logger.error(
+                "Unable to get the Sigma rule",
+                exc_info=True,
+            )
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
+                "Unable to get Sigma rule. Error {0!s}".format(e),
+            )
+
+        if not rule:
+            abort(HTTP_STATUS_CODE_NOT_FOUND, "No rule found with this ID.")
+        return_rules = []
+
+        assert isinstance(rule, SigmaRule)
+
+        return_rules.append(_enrich_sigma_rule_object(rule=rule))
+
+        return jsonify({"objects": return_rules, "meta": {}})
+
+    @login_required
+    def delete(self, rule_uuid):
+        """Handles DELETE request to the resource.
+
+        Handels DELETE API calls to /sigmarule/<string:rule_uuid>/ where the
+        rule_uuid is the primaray way to identify the rule.
+
+        Args:
+            rule_uuid: uuid of the rule
+
+        Returns:
+            HTTP_STATUS_CODE_NOT_FOUND if rule not found
+            HTTP_STATUS_CODE_OK if rule deleted
+        """
+
+        rule = SigmaRule.query.filter_by(rule_uuid=rule_uuid).first()
+
+        if not rule:
+            abort(
+                HTTP_STATUS_CODE_NOT_FOUND,
+                "No rule found with rule_uuid.{0!s}".format(rule_uuid),
+            )
+
+        db_session.delete(rule)
+        db_session.commit()
+
+        return HTTP_STATUS_CODE_OK
+
+    @login_required
+    def post(self, rule_uuid=None):
+        """Handles POST request to the resource.
+        Handels POST API calls to /sigmarule/<string:rule_uuid>/ where the
+        rule_uuid is the primaray way to identify the rule.
+        The remaining attributes of the rule are provided in request itself.
```

Comment:
This is still unclear. Does POST create a new SigmaRule? Does it update an existing one? What happens if rule_uuid is not provided? What happens when it's different than the rule_uuid that's provided in the body of the POST request?

As a developer somehow familiar with flask, it's pretty clear from the code that this handles post requests, but what's not clear the "business logic" behind it.